presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -340,6 +350,20 @@ else if (columnStatistics.getStatsData().isSetBinaryStats()) {
         return Optional.of(new BigDecimal(new BigInteger(decimal.getUnscaled()), decimal.getScale()));
     }
 
+    /**
+     * Metastore stores NDV considering null as a distinct value
+     */
+    private static OptionalLong fromMetastoreDistinctValuesCount(OptionalLong distinctValuesCount, OptionalLong nullsCount)
+    {
+        if (distinctValuesCount.isPresent() && nullsCount.isPresent()) {
+            if (distinctValuesCount.getAsLong() > 0 && nullsCount.getAsLong() > 0) {
+                return OptionalLong.of(distinctValuesCount.getAsLong() - 1);
+            }
+            return distinctValuesCount;
+        }
+        return OptionalLong.empty();
```

Comment:
Now when `distinctValuesCount.isPresent() && !nullsCount.isPresent()` we return absent NDV, while previously we returned NDV with a an error. Maybe it is better to return the something with error than nothing.

Reply:
IRL both `distinctValuesCount` and `nullsCount` are required fields in the HiveMetastore. Although we check whether these values are set, there is a hard check inside every `*StatData` class that both of those fileds are set. Technically it looks like this API is free to be changed, but i don't think that is gonna be an issue, since we are always going to compute `distinctValuesCount` and `nullsCount` in pair in Presto.

Reply:
You're correct. But a person reading this code in the future will need to follow your path to check these things to know this code is correct.

Please replace `return OptionalLong.empty();` with `return distinctValuesCount;` (and remove `return distinctValuesCount;` above)

Reply:
I still think that returning `distinctValuesCount` as is is incorrect. But since it doesn't matter a lot i'm fine with that.